power_grid_model_c/power_grid_model/include/power_grid_model/common/common.hpp

```diff
@@ -30,6 +30,16 @@ struct Idx2D {
     friend constexpr bool operator==(Idx2D x, Idx2D y) = default;
 };
 
+struct symmetric_t {};
+struct asymmetric_t {};
+
+template <typename T>
+concept symmetry_tag = std::derived_from<T, symmetric_t> || std::derived_from<T, asymmetric_t>;
+
+template <symmetry_tag T> constexpr bool is_symmetric = std::derived_from<T, symmetric_t>;
```

Comment:
Follow convension of `type_traits`, use `is_symmetric_v` for meta function of value.

Reply:
Define first a struct `is_symmetric` which derived from `std::true_type` or `std::false_type` depending on the template argument. Then use variable template.

```c++
template <symmetry_tag T>
struct is_symmetric;

template<>
struct is_symmetric<symmetric_t> : std::true_type {};
template<>
struct is_symmetric<asymmetric_t> : std::false_type {};

template <symmetry_tag T> constexpr bool is_symmetric_v = is_symmetric<T>::value;
```

Reply:
what is the reason you prefer type traits over concept syntax?

Reply:
ok, will use the following syntax then 😄 

```cpp
template <symmetry_tag T> constexpr bool is_symmetric_v = std::derived_from<T, symmetric_t>;
```